src/libsystemd/sd-journal/journal-def.h

```diff
@@ -117,7 +117,10 @@ struct HashTableObject {
 struct EntryArrayObject {
         ObjectHeader object;
         le64_t next_entry_array_offset;
-        le64_t items[];
+        union {
+                le64_t regular[0];
+                le32_t compact[0];
```

Comment:
i think i asked this in a previous review, but i forgot: why specify the size here? why not leave it empty? i.e. `[]` rather than `[0]`?

Reply:
Variable length arrays in unions are not standard C, but do work in GCC this way. Just try to define a union containing an array without size and see if it compiles.